src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -719,6 +736,48 @@ private void SearchPackagesAndRefreshUpdateCount(bool useCacheForUpdates)
             .FileAndForget(TelemetryUtility.CreateFileAndForgetEventName(nameof(PackageManagerControl), nameof(SearchPackagesAndRefreshUpdateCount)));
         }
 
+        // Check if user is in A/B experiment. Also return true if environment variable RecommendNuGetPackages is set to 1.
+        public bool IsRecommenderFlightEnabled()
+        {
+            return _forceRecommender || ExperimentationService.Default.IsCachedFlightEnabled("nugetrecommendpkgs");
+        }
+
+        private async Task<(IPackageFeed mainFeed, IPackageFeed recommenderFeed)> GetPackageFeedsAsync(string searchText, PackageLoadContext loadContext)
+        {
+            // only make recommendations when
+            //   the single source repository is nuget.org,
+            //   the package manager was opened for a project, not a solution,
+            //   this is the Browse tab,
+            //   and the search text is an empty string
+            _recommendPackages = false;
+            if (loadContext.IsSolution == false
+                && _topPanel.Filter == ItemFilter.All
+                && searchText == string.Empty
+                && loadContext.SourceRepositories.Count() == 1
+                && Model.Context.Projects.Count() == 1
```

Comment:
this is unnecessary with the IsSolution check. 
No model that's not solution based should contain more than 1 project.

Reply:
removed